packages/ui/src/views/user-management/components/dialogs/components/create-user/types.ts

```diff
@@ -0,0 +1,15 @@
+import { newUserSchema } from '@/views/user-management/components/dialogs/components/create-user/schemas'
+import { z } from 'zod'
+
+export type NewUserFields = z.infer<typeof newUserSchema>
+
+export interface ICreateUserDialogProps {
+  handleCreateUser: (data: NewUserFields) => void
+  open: boolean
+  onClose: () => void
+}
+
+export interface ICreateUserFormProps {
```

Comment:
I'm really not a fan of the `I` prefix on interfaces. For me, the `Props` suffix is indication enough that it is a type.
```suggestion
export interface CreateUserDialogProps {
  handleCreateUser: (data: NewUserFields) => void
  open: boolean
  onClose: () => void
}

export interface CreateUserFormProps {
```

I also prefer when the component props interfaces live with their respective components. Easier to keep them in sync and see at a glance.